src/lib/l1/ECL_L1_Pos_Controller.cpp

```diff
@@ -264,11 +264,6 @@ ECL_L1_Pos_Controller::navigate_loiter(const Vector2f &vector_A, const Vector2f
 	/* calculate velocity on circle / along tangent */
 	float tangent_vel = xtrack_vel_center * loiter_direction;
 
-	/* prevent PD output from turning the wrong way */
-	if (tangent_vel < 0.0f) {
```

Comment:
An alternative to removing this code block that should be evaluated is replacing this line with

if (tangent_vel < 0.0f && _circle_mode) {

So that the protection against getting trapped in a reverse direction loiter is preserved

Reply:
Changing it to `if (tangent_vel < 0.0f && _circle_mode) {` seems to do the trick:

With it:
![image](https://user-images.githubusercontent.com/26798987/95306526-ce51b300-0887-11eb-90ab-c03a2e7926f6.png)
Without:
![image](https://user-images.githubusercontent.com/26798987/95308993-dd863000-088a-11eb-98e9-3ed13ba7e519.png)

I've also tested with reversing the loiter direction. Entry looks awful (guess that's no big surprise), but it consistently ends up doing it the circle in the correct direction.